### PR TITLE
[Documentation] Improved analog QUBO tutorial

### PR DESCRIPTION
**Pull request by madagra**

### CODE REVIEW

File: docs/digital_analog_qc/analog-qubo.md [hunk 1]
```diff
@@ -98,109 +92,102 @@ Q = np.array(
         [5.42015853, 0.85604541, 0.17675796, 0.32306662, -10.0],
     ]
 )
+
+# Cost function for a single bitstring
+def cost_bitstring(bitstring, qubo_mat):
```
Comment (Roland-djee) [inline docs/digital_analog_qc/analog-qubo.md:97]: Do you bother about type annotations ?

File: docs/digital_analog_qc/analog-qubo.md [hunk 2]
```diff
@@ -98,109 +92,102 @@ Q = np.array(
         [5.42015853, 0.85604541, 0.17675796, 0.32306662, -10.0],
     ]
 )
+
+# Cost function for a single bitstring
```
Comment (Roland-djee) [inline docs/digital_analog_qc/analog-qubo.md:96]: Maybe add a formula ?

File: docs/digital_analog_qc/analog-qubo.md [hunk 3]
```diff
@@ -98,109 +92,102 @@ Q = np.array(
...
 ```
 
-The analog circuit is composed of two global rotations per layer.  The first
-rotation corresponds to the mixing Hamiltonian and the second one to the
-embedding Hamiltonian in the QAOA algorithm. Subsequently, there is an Ising interaction
-term to emulate the analog circuit. Please note that the Rydberg level is set to 70. We
-initialize the weighted register graph from the QUBO definition similarly to what is
-done in Pulser, and set the device specs with the updated Rydberg level.
+The QAOA algorithm needs a variational quantum circuit with optimizable parameters.
+We use a fully analog circuit composed of two global rotations per layer on
```
Comment (Roland-djee) [inline docs/digital_analog_qc/analog-qubo.md:109]:
```suggestion
For that purpose, we use a fully analog circuit composed of two global rotations per layer on
```

File: docs/digital_analog_qc/analog-qubo.md [hunk 4]
```diff
@@ -98,109 +92,102 @@ Q = np.array(
...
+different axes of the Bloch sphere.
+The first rotation corresponds to the mixing Hamiltonian and the second one to the
+embedding Hamiltonian (given by the register coordinates and equivalent to a
```
Comment (Roland-djee) [inline docs/digital_analog_qc/analog-qubo.md:112]:
```suggestion
embedding Hamiltonian (given by the register coordinates and equivalent to a ...)
```

I find this part in parenthesis unclear. Not sure if that relates only to the embedding Hamiltonian or the mixing one or both ?

File: docs/digital_analog_qc/analog-qubo.md [hunk 5]
```diff
@@ -98,109 +92,102 @@ Q = np.array(
...
-    QuantumCircuit(reg, block),
-    backend="pyqtorch", diff_mode='gpsr')
+model = QuantumModel(circuit)
 initial_counts = model.sample({}, n_shots=1000)[0]
 
 print(f"initial_counts = {initial_counts}") # markdown-exec: hide
 ```
 
-Then, the loss function is defined by averaging over the evaluated bitstrings.
+Finally, we can proceed to the variational optimization. The cost function
```
Comment (Roland-djee) [inline docs/digital_analog_qc/analog-qubo.md:146]:
```suggestion
Finally, we can proceed with the variational optimization. The cost function
```

File: docs/digital_analog_qc/analog-qubo.md [hunk 6]
```diff
@@ -98,109 +92,102 @@ Q = np.array(
...
+defined above is based on bitstring and thus non differentiable. We use Qadence
```
Comment (Roland-djee) [inline docs/digital_analog_qc/analog-qubo.md:147]:
```suggestion
defined above is derived from bitstring computations and therefore non differentiable. We use Qadence
```

File: docs/digital_analog_qc/analog-qubo.md [hunk 7]
```diff
@@ -98,109 +92,102 @@ Q = np.array(
...
+ML facilities to run some gradient-free optimization based on the
```
Comment (Roland-djee) [inline docs/digital_analog_qc/analog-qubo.md:148]:
```suggestion
ML facilities to run gradient-free optimizations using the
```

**Review by Roland-djee — APPROVED** [review]
Feel free to disregard suggestions if you feel them inappropriate.

File: docs/digital_analog_qc/analog-qubo.md (hunk 4 above)
Comment (madagra) [reply docs/digital_analog_qc/analog-qubo.md:100]: Only the embedding. I will clarify it.

File: docs/digital_analog_qc/analog-qubo.md [hunk 8]
```diff
@@ -1,50 +1,46 @@
 !!! warning
     Tutorial to be updated
```
Comment (jpmoutinho) [inline docs/digital_analog_qc/analog-qubo.md:2]:
```suggestion
```

**Review by jpmoutinho — APPROVED** [review]
Thanks @madagra!

File: docs/digital_analog_qc/analog-qubo.md (hunk 2 above)
Comment (madagra) [reply docs/digital_analog_qc/analog-qubo.md]: No need, it is trivial and there is already ample reference for the reader to check it out by themselves.

**Pull request merged**